### PR TITLE
Recurring fix

### PR DESCRIPTION
**Pull request by muhamadazmy**
- Period and tasks must be read from the service hrd file. Not the state.
- Last run time must be tracked in the state file.

### CODE REVIEW

File: jscockpit/ays_bot/recurring.py [hunk 1]
```diff
@@ -23,23 +23,29 @@ def stop(self):
...
+                        now = j.data.time.epoch
                         if last is None or now > (last + sec):
                             rq = self.bot.schedule_action(
                                 action=action_name,
                                 repo=service.aysrepo.name,
                                 role=service.role,
                                 instance=service.instance,
                                 force=True)
-                            self._services_reccurring[service][action_name]['last'] = now
+                            service.state.recurring[action_name] = now
```
Comment (zaibon) [inline jscockpit/ays_bot/recurring.py]: value recurring dict should contain bot the period and the last execution time.
See https://github.com/Jumpscale/jumpscale_core8/blob/master/lib/JumpScale/baselib/atyourservice/ServiceState.py#L230

File: jscockpit/ays_bot/recurring.py [hunk 2]
```diff
@@ -59,8 +65,11 @@ def _load_aysrepos(self, repos=[]):
 
         for repo in repos:
             for service in repo.findServices():
-                for action_name, period in service.state.recurring.items():
-                    if service not in self._services_reccurring:
-                        self._services_reccurring[service] = {}
-                    self._services_reccurring[service][action_name] = {'period': period[0], 'last': None}
+                tasks = service.hrd.getDictFromPrefix('recurring')
```
Comment (zaibon) [inline jscockpit/ays_bot/recurring.py]: period should be read from the state, not the service.hrd

File: jscockpit/ays_bot/recurring.py (hunk 1 above)
Comment (muhamadazmy) [inline jscockpit/ays_bot/recurring.py]: That doesn't make sense, the period is not a state. It's a configuration value. Hence, when you need to access it you can always get it from the hrd files. Unlike the lastrun time which should be tracked as a state. I think the ServiceState.py of atyourservice is what need to be changed.

File: jscockpit/ays_bot/recurring.py (hunk 2 above)
Comment (muhamadazmy) [inline jscockpit/ays_bot/recurring.py]: what if I changed the value of the service.hrd file ?

File: jscockpit/ays_bot/recurring.py [hunk 3]
```diff
@@ -23,23 +23,29 @@ def stop(self):
...
                         sec = j.data.time.getDeltaTime(info['period'])
-                        last = info['last']
+                        last = service.state.recurring.get(action_name, None)
+                        if not isinstance(last, int):
+                            # backward compatibility with old format.
+                            last = None
 
+                        # this loop can start a lot of action, to make sure our times are in sync, we get the now
+                        # only when we need it.
+                        now = j.data.time.epoch
```
Comment (FastGeert) [inline jscockpit/ays_bot/recurring.py]: j.data.time.epoch is a property?
I would expect that it is a function.

**Pull request closed without merging**